### PR TITLE
Created new API version for Monitor Action Groups

### PR DESCRIPTION
**Pull request by ritwik8119**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ritwik8119** [comment]
Please note that the new changes for this PR are from commit ID 2eeca8b onwards. The previous commits are from earlier changes that have already been merged.

**Comment by zhenglaizhang** [comment]
> Please note that the new changes for this PR are from commit ID [2eeca8b](https://github.com/Azure/azure-rest-api-specs/commit/2eeca8be5d74e60833543722b21d26a2b6b664dd) onwards. The previous commits are from earlier changes that have already been merged.

thanks, usually we care about the PR file diff, not the commits, I suppose the PR diff is actual your change. 
please help fix  LintDiff issues.

File: specification/common-types/resource-management/v2/types.json [hunk 1]
```diff
@@ -638,6 +638,16 @@
     }
   },
   "parameters": {
+    "NotificationIdParameter": {
```
Comment (zhenglaizhang) [inline specification/common-types/resource-management/v2/types.json:641]: the types.json is shared across all RPs, and this is not a good place if you just add one parameter that is needed by your RP. 
btw, what's NotificationIdParameter meaning?

File: specification/common-types/resource-management/v2/types.json [hunk 2]
```diff
@@ -659,7 +669,7 @@
       "in": "path",
       "required": true,
       "type": "string",
-      "description": "The name of the resource group. The name is case insensitive.",
+      "description": "The name of the resource group.",
```
Comment (zhenglaizhang) [inline specification/common-types/resource-management/v2/types.json:672]: any reason to remove the sentence?

**Review by zhenglaizhang — CHANGES_REQUESTED** [review]

File: specification/common-types/resource-management/v2/types.json (hunk 1 above)
Comment (thomasp98296) [reply specification/common-types/resource-management/v2/types.json:641]: Hi thanks for the review. I will remove this it from this file as I only use it in one API I touch.

File: specification/common-types/resource-management/v2/types.json (hunk 2 above)
Comment (thomasp98296) [reply specification/common-types/resource-management/v2/types.json:672]: Hi there was a conflict error where the description is different between this and the other version. I am fairly new to this but I believe I either have to fix the description here, or wherever they appear in other files. Given this is only a description, and that changing it here or at other places (I went ahead and fix it here as it is just one place).

**Comment by thomasp98296** [comment]
> thanks, usually we care about the PR file diff, not the commits, I suppose the PR diff is actual your change.
> please help fix LintDiff issues.

Hi we fixed all of the LintDiff errors related to our changes. The errors current in the LintDiff are not related to ours. Please advise. Thank you so much!

File: specification/common-types/resource-management/v2/types.json (hunk 2 above)
Comment (thomasp98296) [reply specification/common-types/resource-management/v2/types.json:662]: @zhenglaizhang could you please advise us what is the next step here? We have addressed all issues related to our changes. Thank you!

**Comment by thomasp98296** [comment]
@zhenglaizhang could you pleaes 

> > Please note that the new changes for this PR are from commit ID [2eeca8b](https://github.com/Azure/azure-rest-api-specs/commit/2eeca8be5d74e60833543722b21d26a2b6b664dd) onwards. The previous commits are from earlier changes that have already been merged.
> 
> thanks, usually we care about the PR file diff, not the commits, I suppose the PR diff is actual your change. please help fix LintDiff issues.

Hi, we resolved all the comments related to our changes. Could you please advise on the next steps? Thank you!

File: specification/common-types/resource-management/v2/types.json (hunk 2 above)
Comment (zhenglaizhang) [reply specification/common-types/resource-management/v2/types.json:662]: do you have the link to the conflict error? I need to check what's the detail.

**Review by zhenglaizhang — CHANGES_REQUESTED** [review]

File: specification/common-types/resource-management/v2/types.json (hunk 2 above)
Comment (thomasp98296) [reply specification/common-types/resource-management/v2/types.json]: @zhenlaizhang, here is the error with the description.
"readme":"monitor/resource-manager/readme.md",
"tag":"package-2021-09",
"details":"Error: '$.parameters.ResourceGroupNameParameter.description' has incompatible values (---The name of the resource group.,
---The name of the resource group. The name is case insensitive.)."

Comment (thomasp98296) [reply specification/common-types/resource-management/v2/types.json]: > do you have the link to the conflict error? I need to check what's the detail.

Could you please advise the next step here? This PR has been in pending for too long. Thanks!

**Comment by ToddKingMSFT** [comment]
Just because your changes might be good in isolation, you still need to get them to play nice with all the other existing swaggers being grouped together in the readme. That means that if you have a description difference, in most cases that means you need to change your own description to conform with the already existing description.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 3]
```diff
@@ -0,0 +1,1120 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Patch an action group": {
+            "$ref": "./examples/patchActionGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/notifications": {
```
Comment (RamyasreeChakka) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:213]: What is the purpose of this API? As there is no providers information in the API, this API will not be routed to your RP. This API sounds incorrect to me.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 4]
```diff
@@ -0,0 +1,1120 @@
...
+          "default": {
+            "description": "An error occurred while sending the test notifications",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/notifications/{notificationId}": {
```
Comment (RamyasreeChakka) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:250]: What is the purpose of this API? As there is no providers information in the API, this API will not be routed to your RP. This API sounds incorrect to me.

**Comment by RamyasreeChakka** [comment]
@ritwik8119  - can you address my comments?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 5]
```diff
@@ -0,0 +1,1120 @@
...
+    "ActionGroup": {
+      "description": "An Azure action group.",
+      "type":"object",
+      "properties": {
+        "groupShortName": {
+          "type": "string",
+          "maxLength": 12,
+          "description": "The short name of the action group. This will be used in SMS messages."
+        },
+        "enabled": {
```
Comment (RamyasreeChakka) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:604]: >enabled [](http://example.com/codeflow?start=9&length=7)

Replace boolean properties with better enum
Check 'Replace boolean/switch properties with better enum' in [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

Comment (ritwik8119) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:604]: Reading the docs, it seems like a better enum is preferred for bools in general to make it future-proof and allow multiple values if needed, but in this case, the field just indicates if the action is enabled or not, and will always have a True/False value. Is a string enum really needed in this case?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json (hunk 3 above)
Comment (thomasp98296) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:213]: Hi Ramyasree, I am fairly new to this. Is the provider required? The API is intended to be used without the provider in both of the cases. However if it is required, then I can add it. Could you confirm if the provider is required? Thanks!

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 6]
```diff
@@ -0,0 +1,1120 @@
...
+          "201": {
+            "description": "A new action group was successfully created.",
+            "schema": {
+              "$ref": "#/definitions/ActionGroupResource"
+            }
+          },
+          "default": {
+            "description": "An error occurred and the action group could not be created or updated.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:87]: Could refactor to reference ErrorResponse from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 7]
```diff
@@ -0,0 +1,1120 @@
...
+          "description": "The context id type"
+        }
+      }
+    },
+    "ActionGroupResource": {
+      "description": "An action group resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResource"
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:567]: Recommend refactoring to reference TrackedResource from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json to ensure you are following the standard resource patterns

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json [hunk 8]
```diff
@@ -0,0 +1,1120 @@
...
+  "parameters": {
+    "NotificationIdParameter": {
+      "name": "notificationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The notification id",
+      "x-ms-parameter-location": "method"
+    },
+    "ResourceGroupNameParameter": {
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json:1103]: Can use ResourceGroupNameParameter from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json

**Comment by TimLovellSmith** [comment]
> Just because your changes might be good in isolation, you still need to get them to play nice with all the other existing swaggers being grouped together in the readme. That means that if you have a description difference, in most cases that means you need to change your own description to conform with the already existing description.

So the trick to resolving these linter errors is to go into your other api-versions that are referenced from the readme, and update the descriptions to match, is that right?

**Comment by mikekistler** [comment]
I had a quick look at this PR and it looks like there are several API definition files in the `stable/2021-05-01` directory that are not present in the `stable/2021-09-01` directory:

image

This may be what is causing the tooling to flag that there are breaking changes.

**Comment by ritwik8119** [comment]
> @ritwik8119 please help fix the model validation failure https://github.com/Azure/azure-rest-api-specs/pull/16107/checks?check_run_id=3866610947

Fixed. Is there any other Lint/breaking issues that need to be addressed, apart from the ARM API review comments?

**Comment by pilor** [comment]
@zhenglaizhang how did `./examples/postTestNotifications.json` pass ModelValidation?  It is taking in a complex object in the examples parameters but that object is not specified in the operation that references is (`/subscriptions/{subscriptionId}/providers/Microsoft.Insights/notifications`).

**Comment by msyyc** [comment]
approve for python breaking change.

**Review by zhenglaizhang — APPROVED** [review]

**Pull request merged**